lib/fs.js

```diff
@@ -2151,11 +2199,20 @@ fs.mkdtemp = function(prefix, callback) {
   var req = new FSReqWrap();
   req.oncomplete = callback;
 
-  binding.mkdtemp(prefix + 'XXXXXX', req);
+  binding.mkdtemp(prefix + 'XXXXXX', options.encoding, req);
 };
 
-fs.mkdtempSync = function(prefix) {
+fs.mkdtempSync = function(prefix, options) {
+
+  if (!prefix || typeof prefix !== 'string')
+    throw new TypeError('filename prefix is required');
+
+  options = options || {};
+  if (typeof options === 'string')
+    options = {encoding: options};
+  if (typeof options !== 'object')
+    throw new TypeError('"options" must be a string or an object');
   nullCheck(prefix);
 
-  return binding.mkdtemp(prefix + 'XXXXXX');
+  return binding.mkdtemp(prefix + 'XXXXXX', options.encoding);
 };
```

Comment:
Just noting that the type checks for `options` will let through `null`, which will make `options.encoding` fail with a TypeError.  GIGO, IMO - I'll leave it up to you if you want to handle that.